NovaPC_Melhorenvio/app/code/community/NovaPC/Melhorenvio/Helper/Data.php

```diff
@@ -109,6 +109,12 @@ function formatCnpjCpf($value)
 		return preg_replace("/(\d{2})(\d{3})(\d{3})(\d{4})(\d{2})/", "\$1.\$2.\$3/\$4-\$5", $cnpj_cpf);
 	}
 
+	public function isCNPJ($valor){
+		$valor = trim($valor);
+        $valor = str_replace(array(".", ",", "-", "/"), "", $valor);

```

Comment:
Alterar para `preg_replace('/\D/', '', $valor)`. É mais garantido que irá filtrar apenas dígitos.

Reply:
Alteração realizada no commit "Improvements in the function responsible for identifying whether the type of document informed by the buyer is CNPJ"